addons/hw_posbox_homepage/controllers/main.py

```diff
@@ -280,3 +289,71 @@ def clear_six_payment_terminal(self):
         helpers.unlink_file('odoo-six-payment-terminal.conf')
         subprocess.check_call(["sudo", "service", "odoo", "restart"])
         return "<meta http-equiv='refresh' content='0; url=http://" + helpers.get_ip() + ":8069'>"
+
+    @http.route('/hw_proxy/upgrade', type='http', auth='none', )
+    def upgrade(self):
+        commit = subprocess.check_output(["git", "--work-tree=/home/pi/odoo/", "--git-dir=/home/pi/odoo/.git", "log", "-1"]).decode('utf-8').replace("\n", "<br/>")
+        flashToVersion = helpers.check_image()
+        actualVersion = helpers.get_version()
+        if flashToVersion:
+            flashToVersion = '%s.%s' % (flashToVersion.get('major', ''), flashToVersion.get('minor', ''))
+        return upgrade_page_template.render({
+            'title': "Odoo's IoTBox - Software Upgrade",
+            'breadcrumb': 'IoT Box Software Upgrade',
+            'loading_message': 'Updating IoT box',
+            'commit': commit,
+            'flashToVersion': flashToVersion,
+            'actualVersion': actualVersion,
+        })
+
+    @http.route('/hw_proxy/perform_upgrade', type='http', auth='none')
+    def perform_upgrade(self):
+        self.updating.acquire()
+        os.system('/home/pi/odoo/addons/point_of_sale/tools/posbox/configuration/posbox_update.sh')
+        self.updating.release()
+        return 'SUCCESS'
+
+    @http.route('/hw_proxy/get_version', type='http', auth='none')
+    def check_version(self):
+        return helpers.get_version()
+
+    @http.route('/hw_proxy/perform_flashing_create_partition', type='http', auth='none')
+    def perform_flashing_create_partition(self):
+        try:
+            response = subprocess.check_output(['sudo', 'bash', '-c', '. /home/pi/odoo/addons/point_of_sale/tools/posbox/configuration/upgrade.sh; create_partition']).decode().split('\n')[-2]
+            if response == 'Error_Card_Size':
+                raise Exception(response)
+            return 'SUCCESS'
```

Comment:
Wouldn't it be better to make something like: return response('success', status=200)